src/externs/datatables.js

```diff
@@ -26,7 +27,27 @@ jQuery.prototype.DataTable = function(opt_params) {};
 DataTables.prototype.rows = function(opt_params) {};
 
 /**
- * @param {Object=} opt_params
+ * @param {number} params
  * @return {DataTables.Api}
  */
-DataTables.prototype.invalidate = function(opt_params) {};
+DataTables.prototype.button = function(params) {};
+
+/**
+ * @param {Object=} opt_params
+ */
+DataTables.Api.prototype.invalidate = function(opt_params) {};
+
+/**
+ * @param {Object=} opt_params
```

Comment:
Can you typedef DataTables.Api to be Array so that it doesn't prevent your type coersion? Doc mentioned that DataTables.Api is Array-like https://datatables.net/reference/type/DataTables.Api